### PR TITLE
[AL-167] - Android support black fingerprint

### PR DESCRIPTION
**Pull request by Al-Taie**
> Jira Task: [AL-167](https://npi.atlassian.net/browse/AL-167)

[AL-167]: https://npi.atlassian.net/browse/AL-167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: fingerprint/src/main/java/com/futronictech/Scanner.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    @JvmName("SetLogOptions")
+    external fun setLogOptions(i: Int, i2: Int, str: String?): Boolean
+
+    @JvmName("SetNewAuthorizationCode")
+    external fun setNewAuthorizationCode(bytes: ByteArray?): Boolean
+
+    @JvmName("SetOptions")
+    external fun setOptions(i: Int, i2: Int): Boolean
+
+    val imageWidth: Int
```
Comment (RoonyShekho09) [inline fingerprint/src/main/java/com/futronictech/Scanner.kt:93]: It appears that this is not used, same with a lot of other functions and variables, are they going to be used in the future?

File: fingerprint/src/main/java/com/fingerprint/scanner/FingerprintScanner.kt [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 
 import android.hardware.usb.UsbDevice
+import com.fingerprint.manager.FingerprintDeviceInfo
 import com.fingerprint.utils.ScannedImageType
 
 
 internal interface FingerprintScanner {
-    fun tunOffLed()
+    val deviceInfo: FingerprintDeviceInfo
+    fun tunOffLed() = Unit
```
Comment (RoonyShekho09) [inline fingerprint/src/main/java/com/fingerprint/scanner/FingerprintScanner.kt]: You missed the r

File: fingerprint/src/main/java/com/fingerprint/manager/FingerprintManagerImpl.kt [hunk 3]
```diff
@@ -174,19 +183,27 @@ internal class FingerprintManagerImpl(
...
-        eventsFlow.emit(if (captureIndex == 0) FingerprintEvent.PlaceFinger else FingerprintEvent.KeepFinger)
+        val isFirstCapture = captureIndex == 0
+        if (isFirstCapture)
+            eventsFlow.emit(FingerprintEvent.PlaceFinger)
+        else
+            eventsFlow.emit(FingerprintEvent.KeepFinger)
 
         if (!captureImage()) return
 
         if (!getImageData()) return
```
Comment (RoonyShekho09) [inline fingerprint/src/main/java/com/fingerprint/manager/FingerprintManagerImpl.kt]: Why not put both of them in one statement?

File: fingerprint/src/main/java/com/fingerprint/scanner/HfSecurityFingerprintScanner.kt [hunk 4]
```diff
@@ -57,45 +69,18 @@ internal class HfSecurityFingerprint(
...
-
-    override fun getImageData(): ByteArray? {
+    override suspend fun getImageBytes(): ByteArray? {
         val imageData = ByteArray(imageType.size)
         val result = if (imageType == ScannedImageType.Normal)
             getImageData(imageData)
         else
             getImageDataExtra(imageData)
-        return if (result) imageData else null
+        return if (result) imageData.convertImageDataToBitmapArray(
```
Comment (RoonyShekho09) [inline fingerprint/src/main/java/com/fingerprint/scanner/HfSecurityFingerprintScanner.kt]: This is not related to the PR but you can choose a better name than "result", something that make it appear that it's boolean, like isResultSuccess

**Review by RoonyShekho09 — COMMENTED** [review]
Good work, I wrote some comments

File: fingerprint/src/main/java/com/futronictech/Scanner.kt (hunk 1 above)
Comment (Al-Taie) [reply fingerprint/src/main/java/com/futronictech/Scanner.kt:93]: These functions are used in the c++ api

**Review by RoonyShekho09 — APPROVED** [review]

**Pull request merged**